### DISCUSSION TITLE
How can I make an observable based on 2 others, but emit only when the 2 observables emit new values

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [ReactiveX/rxjs#7112] [Q&A] [answered]
How can I make an observable based on 2 others observables, but emit only when the 2 observables emit new values (regardless of whether it is equal to the previous value emitted), similar to `combineLatest`, but I want it to emit only when the 2 observables emit new values

input:
```ts
import { of, delay, concatMap } from 'rxjs';

const observable1$ = of('a', 'a', 'b').pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);
const observable2$ = of(1, 2, 3, 4).pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);
```
That is:
```
observable1$:  'a'--'a'--'b'-----
observable2$:  '1'--'2'--'3'--'4'
```
Expexted output:
```batch
["a", "1"]
["a", "2"]
["b", "3"]
```

With `combineLatestWith` i got other output

```ts
import { combineLatestWith, of, delay, concatMap } from 'rxjs';

const observable1$ = of('a', 'a', 'b').pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);
const observable2$ = of('1', '2', '3', '4').pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);

observable1$.pipe(combineLatestWith(observable2$)).subscribe(console.log);
```
```batch
["a", "1"]
["a", "1"]
["a", "2"]
["b", "2"]
["b", "3"]
["b", "4"]
```

**Comment by XuankangLin** [comment]
Pipe each one using “distinctUntilChanged” then “combineLatest”?

**Reply by MiniSuperDev** [reply]
@XuankangLin  Thanks, but it doesn't work for my case, I already updated the question, because initially I didn't make myself understood, thanks

**Reply by XuankangLin** [reply]
From your updated example, maybe you are looking for `withLatestFrom`?

**Comment by josepot** [comment, marked as answer]
Hi @MiniSuperDev !

If I understood the question correctly, then I think that the operator that you are looking for is [`zip`](https://rxjs.dev/api/index/function/zip):

```ts
import { zip, of, delay, concatMap } from 'rxjs';

const observable1$ = of('a', 'a', 'b').pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);
const observable2$ = of('1', '2', '3', '4').pipe(
  concatMap((v) => of(v).pipe(delay(100)))
);

zip(observabl1, observable2).subscribe(console.log);
```

Just be cautious of backpressure.

**Reply by MiniSuperDev** [reply]
Thanks @josepot, it works as expected.

What do you mean by "backpressure"?

**Reply by josepot** [reply]
> Thanks @josepot, it works as expected.

:tada: if that's the case, then could you please mark the answer as "accepted"?

> 
> What do you mean by "backpressure"?

[Check this out
](https://reactivex.io/documentation/ko/operators/backpressure.html)
